### ISSUE TITLE
GRPC Compatible?

### ISSUE THREAD

**Issue by kferrone** [eclipse/codewind#1575]
So I was trying to set up my first project to work with Codewind. This project is a typescript based GRPC server. I had everything almost exactly like the simple codewind node project which is generated from saying new project. I'm starting to think I can't get it to work because of GRPC and Codewind not being compatible. Is this the case? Can I use Codewind with GRPC or is Codewind expecting a normal HTTP server? If so I would love to see an example of it working.

**Comment by tetchel** [comment]
woops, https://github.com/eclipse/codewind-vscode/pull/362 is not related, bad paste.

I will get a team member to respond, however...

**Comment by johnmcollier** [comment]
Hey @kferrone, thanks for opening an issue!

When you say that you couldn't get it to work, do you know what happened exactly, or what errors (if any) were shown in the UI? 

Generally Codewind expects the projects it deploys to have an exposed HTTP server to communicate with, so I don't think gRPC would work well with Codewind **right now**. But, given its popularity in microservices, it's something worth investigating on our side.

**Comment by elsony** [comment]
This is an existing item for handling apps that doesn't expose http port. One proposed way to resolve it can be found in https://github.com/eclipse/codewind/issues/1269#issuecomment-558347306.  Having said that, as @johnmcollier mentioned, knowing more details will help us to have a better understanding in this specific issue.

**Comment by kferrone** [comment]
So I was using the standard template because I already wrote my own Dockerfile and I'm pretty proficient with Helm. Never heard of Appsody until Codewind gave me an example. 

The issue in the UI is it says the Build failed making the image. However the logs in the Dockerfile say everything worked just fine. In my Docker client (eclipse) I can see the container it started and it says it is working just fine and started. I could even connect my client to it just fine. So this is why I figured it's just codewind is unable to tell if it started or not. 

Since there is a configuration for `health` endpoint I was already wondering how I would even do that with GRPC?

**Comment by johnmcollier** [comment]
@kferrone Yeah, I don't think Codewind's current design would work well for non-HTTP servers at this moment. But I think this is something we should investigate further. 

That being said, the `Build failed` message you got seems like a bug given the docker logs showed it built fine and it was running on your machine. I would have expected the status to be `[Stopped] [Build Succeeded]`, as we wouldn't have been able to ping any HTTP server, but the app did build fine.

Would you be able to share the docker logs from the `codewind-pfe` container? I'm curious to see what made Codewind think the build failed.
